### PR TITLE
Emscripten support

### PR DESCRIPTION
**Pull request by Guillaume227**
I am a total emscripten beginner (picked it up last week for the purpose of marl integration) but I managed to get the marl examples to run in the browser, which is nice. 
I updated the readme with what should be simple steps. Let me know if I have missed anything.

One change worth noting : I had to extend the marl low-level API by adding `marl_main_fiber_init` for the purpose of initializing the first (main) fiber on a given emscripten thread. It reflects the `emscripten_fiber_init_from_current_context` emscripten/fiber primitive. From the emscripten doc: 

> This is needed in order to switch back from a fiber into the thread’s original context.

For all other platforms, that method does nothing.

---- 
I would recommend squashing the commits (not sure if I can do that on my end).

### CODE REVIEW

File: examples/shell.emscripten.html [hunk 1]
```diff
@@ -0,0 +1,194 @@
...
+<div class="spinner" id='spinner'></div>
+<div class="emscripten" id="status">Downloading...</div>
+
+<div class="emscripten">
+    <progress value="0" max="100" id="progress" hidden=1></progress>
+</div>
+
+
+<div class="emscripten_border">
+    <canvas class="emscripten" id="canvas" oncontextmenu="event.preventDefault()" tabindex=-1></canvas>
```
Comment (ben-clayton) [inline examples/shell.emscripten.html:115]: Is this canvas actually used by anything, or can it be removed?

File: src/osfiber_emscripten.cpp [hunk 2]
```diff
@@ -0,0 +1,60 @@
+// Copyright 2019 The Marl Authors.
```
Comment (ben-clayton) [inline src/osfiber_emscripten.cpp:1]: nit: 2023

**Review by ben-clayton — COMMENTED** [review]
Looks great!

File: examples/shell.emscripten.html (hunk 1 above)
Comment (Guillaume227) [reply examples/shell.emscripten.html:115]: I am not sure, my html/javascript is  very, very basic. I was only interesting in putting together some demo without trying to be minimal. Is it worth investing time given it's only an example for people to adapt anyway?

File: src/osfiber_emscripten.cpp (hunk 2 above)
Comment (Guillaume227) [reply src/osfiber_emscripten.cpp]: updated

**Comment by ben-clayton** [comment]
```
Scanning 106 files...
2 errors:
* examples/run_webserver has no license
* examples/shell.emscripten.html has no license
```

Please add the license header to these files, or add a exemption to `license-checker.cfg`

**Comment by Guillaume227** [comment]
done !

File: examples/run_webserver [hunk 3]
```diff
@@ -1,5 +1,19 @@
 #!/usr/bin/env python3
 """
+// Copyright 2023 The Marl Authors.
```
Comment (ben-clayton) [inline examples/run_webserver:3]: Copyright needs to be outside of the quote.
I'd recommend just adding this file to the [`license-checker.cfg` `"exclude"` list](https://github.com/google/marl/blob/9c689c9a88bd8c6a8fe48b01d5ec6b71974282a6/license-checker.cfg#L8-L22):

File: examples/shell.emscripten.html [hunk 4]
```diff
@@ -1,3 +1,17 @@
+// Copyright 2023 The Marl Authors.
```
Comment (ben-clayton) [inline examples/shell.emscripten.html:1]: HTML quote is `<!-- comment -->`, but some browsers apparently can't handle a comment before the `<!doctype html>`.
I'd recommend just adding this file to the [`license-checker.cfg` `"exclude"` list](https://github.com/google/marl/blob/9c689c9a88bd8c6a8fe48b01d5ec6b71974282a6/license-checker.cfg#L8-L22):

**Review by ben-clayton — COMMENTED** [review]
Sorry, one last round of comments.

File: examples/shell.emscripten.html (hunk 1 above)
Comment (Guillaume227) [reply examples/shell.emscripten.html]: ok - I experimented with removing that canvas which still works and makes for a more minimal example. Updated.

File: examples/run_webserver (hunk 3 above)
Comment (Guillaume227) [reply examples/run_webserver]: exemption added

File: examples/shell.emscripten.html (hunk 4 above)
Comment (Guillaume227) [reply examples/shell.emscripten.html]: exemption added

**Comment by ben-clayton** [comment]
Looks great. Thank you for all the changes!

**Pull request merged**